### PR TITLE
NOTIF-701 Unify endpoints query

### PR DESCRIPTION
**Pull request by josejulio**
- We had a query that didn't took into account the active filter Decided to unify both queries into a single one

### CODE REVIEW

File: backend/src/test/java/com/redhat/cloud/notifications/routers/EndpointResourceTest.java [hunk 1]
```diff
@@ -1412,6 +1415,64 @@ void testConnectionCount() {
...
+
+        endpointPage = Json.decodeValue(response.getBody().asString(), EndpointPage.class);
+        assertEquals(stats[0] - stats[1], endpointPage.getMeta().getCount());
+        assertEquals(stats[0] - stats[1], endpointPage.getData().size());
+
+        // Only inactive
+        response = given()
+                .header(identityHeader)
+                .when()
+                .queryParam("active", "true")
```
Comment (gwenneg) [inline backend/src/test/java/com/redhat/cloud/notifications/routers/EndpointResourceTest.java:1463]: Shouldn't this be `false`?

File: backend/src/test/java/com/redhat/cloud/notifications/routers/EndpointResourceTest.java [hunk 2]
```diff
@@ -1412,6 +1415,64 @@ void testConnectionCount() {
...
+    void testActive() {
+        String orgId = "queries-without-type";
+        String username = "user";
+        String identityHeaderValue = TestHelpers.encodeRHIdentityInfo(TestConstants.DEFAULT_ACCOUNT_ID, orgId, username);
+        Header identityHeader = TestHelpers.createRHIdentityHeader(identityHeaderValue);
+        MockServerConfig.addMockRbacAccess(identityHeaderValue, MockServerConfig.RbacAccess.FULL_ACCESS);
+
+        // stats[0] is the count
+        // stats[1] are the inactive ones
+        int[] stats = resourceHelpers.createTestEndpoints(TestConstants.DEFAULT_ACCOUNT_ID, orgId, 10);
```
Comment (gwenneg) [inline backend/src/test/java/com/redhat/cloud/notifications/routers/EndpointResourceTest.java:1428]: With `10`, the helper creates an even number of enabled and disabled endpoints, which doesn't help when assertions are run on `getMeta().getCount()` or `getData().size()`.
```suggestion
        int[] stats = resourceHelpers.createTestEndpoints(TestConstants.DEFAULT_ACCOUNT_ID, orgId, 11);
```

File: backend/src/main/java/com/redhat/cloud/notifications/db/repositories/EndpointRepository.java [hunk 3]
```diff
@@ -149,30 +149,6 @@ public QueryBuilder<Endpoint> getEndpointsQuery(String orgId, @Nullable String n
...
-                .limit(limit)
-                .sort(sort)
-                .build(entityManager::createQuery)
-                .getResultList();
-        loadProperties(endpoints);
-        return endpoints;
-    }
-
-    public Long getEndpointsCount(String orgId, @Nullable String name) {
-        return getEndpointsQuery(orgId, name)
```
[the remark this answers is not in the document]
Comment (josejulio) [reply backend/src/main/java/com/redhat/cloud/notifications/db/repositories/EndpointRepository.java:171]: Not anymore

File: backend/src/test/java/com/redhat/cloud/notifications/routers/EndpointResourceTest.java (hunk 2 above)
Comment (josejulio) [reply backend/src/test/java/com/redhat/cloud/notifications/routers/EndpointResourceTest.java:1428]: oh - that's right - thanks!

Comment (josejulio) [reply backend/src/test/java/com/redhat/cloud/notifications/routers/EndpointResourceTest.java]: Updated to actually run the test correctly - Correctly failed when updated to use 11 endpoints.

**Review by gwenneg — APPROVED** [review]
LGTM

**Pull request merged**